### ISSUE TITLE
Support for short form hex colors for StyleSheets

### ISSUE THREAD

**Issue by arthuralee** [facebook/react-native#376]
The current implementation allows StyleSheet color properties to be set with hex strings or html color strings, for example
```
borderColor: '#d6d7da',
color: 'red'
```
However, short form hex colors do not work:
```
borderColor: '#ccc', // throws 'InvalidColorString'
```
Short form for hex colors should be supported for similarity to CSS.

**Comment by colinramsay** [comment]
Looks like the conversion just checks for a hash (#) and then assumes it'll be a six-character string:

https://github.com/facebook/react-native/blob/master/React/Base/RCTConvert.m#L360

Should be fairly easy to add a length check in there too and behave accordingly.

**Comment by arthuralee** [comment]
Yep. Thanks for the code reference. I've submitted a potential fix at #455.

**Comment by arthuralee** [comment]
Pull request merged internally.
